tiatoolbox/models/architecture/vanilla.py

```diff
@@ -78,6 +78,40 @@ def _get_architecture(
     return model.features
 
 
+def infer_batch(
+    model: nn.Module,
+    batch_data: torch.Tensor,
+    *,
+    device: str = "cpu",
+) -> dict[str, np.ndarray]:
+    """Run inference on an input batch.
+
+    Contains logic for forward operation as well as i/o aggregation.
+
+    Args:
+        model (nn.Module):
+            PyTorch defined model.
+        batch_data (torch.Tensor):
+            A batch of data generated by
+            `torch.utils.data.DataLoader`.
+        device (str):
+            Transfers model to the specified device. Default is "cpu".
+
+    """
+    img_patches_device = batch_data.to(device).type(
+        torch.float32,
+    )  # to NCHW
+    img_patches_device = img_patches_device.permute(0, 3, 1, 2).contiguous()
+
+    # Inference mode
+    model.eval()
+    # Do not compute the gradient (not training)
+    with torch.inference_mode():
+        output = model(img_patches_device)
+    # Output should be a single tensor or scalar
+    return {"probabilities": output.cpu().numpy()}
```

Comment:
In the current `develop` branch, neither `CNNModel`, nor `CNNBackbone` returned dictionaries as output of their `infer_batch()` methods. Also, `CNNModel` currently returns an array, while `CNNBackbone` returns a list with the array. It might be fine, just wanted to highlight this.

`CNNModel`
https://github.com/TissueImageAnalytics/tiatoolbox/blob/76f02f8561313dae786ebc15ab3196bde836a441/tiatoolbox/models/architecture/vanilla.py#L173

`CNNBackbone`
https://github.com/TissueImageAnalytics/tiatoolbox/blob/76f02f8561313dae786ebc15ab3196bde836a441/tiatoolbox/models/architecture/vanilla.py#L270

Reply:
Thanks. We are aware of this. Our preference is to use torch nn models but to generalise for multi modal output we may need dictionaries. This PR is to check if we can move to generic torch models or we will need a sub class.

Reply:
Makes sense.